File: backend/models/call_tracker.go
package models
import "time"

type CallTracker struct {
	ID                 string    `json:"expertId" db:"expert_id"`
	Name               string    `json:"name" db:"name"`
	OrganizationID	   string    `json:"organizationId" db:"organization_id"`
	ProjectID		   string    `json:"projectId" db:"project_id"`
	Favorite           bool      `json:"favorite" db:"favorite"`
	Profession         string    `json:"profession" db:"profession"`
	Company            string    `json:"company" db:"company"`
    CompanyType        string    `json:"companyType" db:"company_type"`
	StartDate    	   time.Time `json:"startDate" db:"start_date"`
	Description        string    `json:"description" db:"description"`
	Geography          string    `json:"geography" db:"geography"`
	Angle              string    `json:"angle" db:"angle"`
	Status             string    `json:"status" db:"status"`
	AIAssessment       int       `json:"aiAssessment" db:"ai_assessment"`
	AIAnalysis         string    `json:"aiAnalysis" db:"ai_analysis"`
	Comments           string    `json:"comments" db:"comments"`
	Availabilities     []Availability    `json:"availabilities" db:"availabilities"`
	ExpertNetworkName  string    `json:"expertNetworkName" db:"expert_network_name"`
	Cost               float64   `json:"cost" db:"cost"`
	ScreeningQuestionsAndAnswers []Question `json:"screeningQuestionsAndAnswers" db:"screening_questions_and_answers"` 
	EmploymentHistory  []Job     `json:"employmentHistory" db:"employment_history"`
	AddedExpertBy      string    `json:"addedExpertBy" db:"added_expert_by"`
	DateAddedExpert    time.Time `json:"dateAddedExpert" db:"date_added_expert"`
	Trends             string    `json:"trends" db:"trends"`
    AddedCallBy        string    `json:"addedCallBy" db:"added_call_by"`
    DateAddedCall      time.Time `json:"dateAddedCall" db:"date_added_call"`
    InviteSent         bool      `json:"inviteSent" db:"invite_sent"`
    MeetingStartDate   time.Time `json:"meetingStartDate" db:"meeting_start_date"`
    MeetingEndDate     time.Time `json:"meetingEndDate" db:"meeting_end_date"`
    PaidStatus         bool      `json:"paidStatus" db:"paid_status"`
    Rating             int      `json:"rating" db:"rating"`
	LinkedInLink       string               `json:"linkedInLink" db:"linkedIn_link"`
}

File: frontend/lib/utils/models/call_tracker.dart
import 'package:admin/utils/models/available_expert.dart';

class CallTracker {
  bool isSelected;
  bool favorite;
  String id;
  String? name;
  String? projectId;
  String? organizationId;
  String? profession;
  String? company;
  String? companyType;
  DateTime? startDate;
  String? description;
  String? geography;
  String? angle;
  String? status;
  int? aiAssessment;
  String? aiAnalysis;
  String? comments;
  List<Availability>? availabilities;
  String? expertNetworkName;
  double? cost;
  List<Question>? screeningQuestionsAndAnswers;
  List<Job>? employmentHistory;
  String? addedExpertBy;
  DateTime? dateAddedExpert;
  String? linkedInLink;
  String? trends;
  String? addedCallBy;
  DateTime? dateAddedCall;
  bool inviteSent;
  DateTime? meetingStartDate;
  DateTime? meetingEndDate;
  bool? paidStatus;
  int? rating;

  CallTracker({
    this.isSelected = false,
    required this.id,
    this.name,
    this.projectId,
    this.favorite = false,
    this.organizationId,
    this.profession,
    this.company,
    this.companyType,
    this.startDate,
    this.description,
    this.geography,
    this.angle,
    this.status,
    this.aiAssessment,
    this.aiAnalysis,
    this.comments,
    this.availabilities,
    this.expertNetworkName,
    this.cost,
    this.screeningQuestionsAndAnswers,
    this.employmentHistory,
    this.addedExpertBy,
    this.dateAddedExpert,
    this.trends,
    this.addedCallBy,
    this.dateAddedCall,
    this.inviteSent = false,
    this.meetingStartDate,
    this.meetingEndDate,
    this.paidStatus,
    this.rating,
    this.linkedInLink,
  });

  factory CallTracker.fromJson(Map<String, dynamic> json) {
    return CallTracker(
      isSelected: json['isSelected'] ?? false,
      id: json['expertId'],
      name: json['name'],
      projectId: json['projectId'],
      favorite: json['favorite'] ?? false,
      organizationId: json['organizationId'],
      profession: json['profession'],
      company: json['company'],
      companyType: json['companyType'],
      startDate:
          json['startDate'] != null ? DateTime.parse(json['startDate']) : null,
      description: json['description'],
      geography: json['geography'],
      angle: json['angle'],
      status: json['status'],
      aiAssessment: json['aiAssessment'],
      aiAnalysis: json['aiAnalysis'],
      comments: json['comments'],
      availabilities: json['availabilities'] != null
          ? List<Availability>.from(
              json['availabilities'].map((x) => Availability.fromJson(x)))
          : null,
      expertNetworkName: json['expertNetworkName'],
      cost: json['cost']?.toDouble(),
      screeningQuestionsAndAnswers: json['screeningQuestionsAndAnswers'] != null
          ? List<Question>.from(json['screeningQuestionsAndAnswers']
              .map((x) => Question.fromJson(x)))
          : null,
      employmentHistory: json['employmentHistory'] != null
          ? List<Job>.from(
              json['employmentHistory'].map((x) => Job.fromJson(x)))
          : null,
      addedExpertBy: json['addedExpertBy'],
      dateAddedExpert: json['dateAddedExpert'] != null
          ? DateTime.parse(json['dateAddedExpert'])
          : null,
      trends: json['trends'],
      addedCallBy: json['addedCallBy'],
      dateAddedCall: json['dateAddedCall'] != null
          ? DateTime.parse(json['dateAddedCall'])
          : null,
      inviteSent: json['inviteSent'] ?? false,
      meetingStartDate: json['meetingStartDate'] != null
          ? DateTime.parse(json['meetingStartDate'])
          : null,
      meetingEndDate: json['meetingEndDate'] != null
          ? DateTime.parse(json['meetingEndDate'])
          : null,
      paidStatus: json['paidStatus'],
      rating: json['rating'],
      linkedInLink: json['linkedInLink'],
    );
  }

  factory CallTracker.defaultCall() {
    return CallTracker(
      id: 'default_id',
      isSelected: false,
      favorite: false,
      name: 'Default Name',
      projectId: 'default_project_id',
      organizationId: 'default_organization_id',
      profession: 'Default Profession',
      company: 'Default Company',
      companyType: 'Default Company Type',
      startDate: DateTime.now(),
      description: 'Default Description',
      geography: 'Default Geography',
      angle: 'Default Angle',
      status: 'Default Status',
      aiAssessment: 0,
      aiAnalysis: 'Default AI Analysis',
      comments: 'Default Comments',
      availabilities: [],
      expertNetworkName: 'Default Expert Network Name',
      cost: 0.0,
      screeningQuestionsAndAnswers: [],
      employmentHistory: [],
      addedExpertBy: 'Default Added Expert By',
      dateAddedExpert: DateTime.now(),
      trends: 'Default Trends',
      addedCallBy: 'Default Added Call By',
      dateAddedCall: DateTime.now(),
      inviteSent: false,
      meetingStartDate: DateTime.now(),
      meetingEndDate: DateTime.now(),
      paidStatus: false,
      rating: 0,
      linkedInLink: 'default_linkedin_link',
    );
  }
}
